packages/material-ui/src/Portal/Portal.js

```diff
@@ -1,82 +1,49 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
 import PropTypes from 'prop-types';
-import ownerDocument from '../utils/ownerDocument';
+import { useForkRef } from '../utils/reactHelpers';
 import { exactProp } from '@material-ui/utils';
 
-function getContainer(container, defaultContainer) {
+function getContainer(container) {
   container = typeof container === 'function' ? container() : container;
-  return ReactDOM.findDOMNode(container) || defaultContainer;
+  return ReactDOM.findDOMNode(container);
 }
 
-function getOwnerDocument(element) {
-  return ownerDocument(ReactDOM.findDOMNode(element));
-}
+const useEnhancedEffect = typeof window !== 'undefined' ? React.useLayoutEffect : React.useEffect;
 
 /**
  * Portals provide a first-class way to render children into a DOM node
  * that exists outside the DOM hierarchy of the parent component.
  */
-class Portal extends React.Component {
-  componentDidMount() {
-    this.setMountNode(this.props.container);
-
-    // Only rerender if needed
-    if (!this.props.disablePortal) {
-      this.forceUpdate(this.props.onRendered);
+const Portal = React.forwardRef(function Portal(props, ref) {
+  const { children, container, disablePortal, onRendered } = props;
+  const [mountNode, setMountNode] = React.useState(null);
+  const childRef = React.useRef(null);
```

Comment:
Just curious, do these explicitly need to be null?

Reply:
Reduces the number of overloads. React will set it to `null` when cleaning up so you end up with `T | undefined | null` if you don't initialize it. But I don't see a problem if you just compare weakly against nullish i.e. `ref.current != null`.

Reply:
Sebastian is right on spot. I have noticed the null / undefined difference in the tests that assert the ref values. It's inconsistent with the codebase. I'm revering back to `const childRef = React.useRef(null);`. Thanks

Reply:
Hum, actually, I see a couple of different usage in the codebase:
- `React.useRef(undefined);` x 1
- `React.useRef(null)` x 10
- `React.useRef()` x38

Reply:
Yeah, that’s why I asked :P As long as we stick to one I’m fine with either blank or null

Reply:
If I understand Sebastian point, it's better in the demos to use null to reduce the number of types to write.

Reply:
I'm going with: null for the demos, nothing for the core.

Reply:
For refs to components I would prefer `useRef(null)`. This is equivalent to the old `createRef` style prevalent in classes:
```js
const createdRef       = React.createRef(); //    createdRef.current === null
const usedRef          = React.useRef(null); //      usedRef.current === null
const usedUndefinedRef = React.useRef(); // usedUndefinedRef.current === undefined
```

Reply:
OK, good for me. The bundle size increase is negligeable. I will update the pull request to match createRef. It's simpler.